File: benchmarks/r/05_incidence.R
# Benchmark 05: Incidence
# R equivalent: IncidencePrevalence::estimateIncidence()

source("benchmarks/r/00_helpers.R")
library(IncidencePrevalence)
cat("=== 05: Incidence ===\n")
t0 <- proc.time()

con <- dbConnect(duckdb(), dbdir = DB_PATH)
cdm <- cdmFromCon(con = con, cdmSchema = "main", writeSchema = "main", cdmName = "synthea_1k")

# Generate denominator
cdm <- generateDenominatorCohortSet(
  cdm = cdm,
  name = "denominator",
  daysPriorObservation = 0
)

# Generate outcome cohort for coronary arteriosclerosis
cdm <- generateConceptCohortSet(
  cdm = cdm,
  conceptSet = list(coronary_artery = 317576),
  name = "outcome_cohort"
)

inc <- estimateIncidence(
  cdm = cdm,
  denominatorTable = "denominator",
  outcomeTable = "outcome_cohort",
  interval = "years",
  repeatedEvents = FALSE
)

save_result(as.data.frame(inc), "05_incidence")
elapsed <- (proc.time() - t0)[["elapsed"]]
save_timing("05_incidence", elapsed)
cdmDisconnect(cdm)
cat("Done in", elapsed, "seconds\n")

File: benchmarks/python/05_incidence.py
"""Benchmark 05: Incidence — omopy.incidence.estimate_incidence()"""

from helpers import Timer, connect_cdm, save_result, save_timing

from omopy.connector import generate_concept_cohort_set
from omopy.generics import Codelist
from omopy.incidence import estimate_incidence, generate_denominator_cohort_set

print("=== 05: Incidence ===")
t = Timer()
cdm = connect_cdm()

cdm = generate_concept_cohort_set(
    cdm, Codelist({"coronary_artery": [317576]}), name="outcome_cohort"
)
cdm = generate_denominator_cohort_set(cdm, name="denominator", days_prior_observation=0)

result = estimate_incidence(
    cdm,
    denominator_table="denominator",
    outcome_table="outcome_cohort",
    interval="years",
    repeated_events=False,
)

save_result(result.data, "05_incidence")
save_timing("05_incidence", t.elapsed())
print(f"Done in {t.elapsed():.2f} seconds")
